gdk4-win32/tests/abi.rs

```diff
@@ -0,0 +1,214 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
```

Comment:
I think the abi tests can't really work under windows, maybe we should look into adding an option to disable generating them or maybe they can be useful for cross compiling cases

Reply:
currently cargo test fails with:
```
User@WinDev2202Eval UCRT64 ~/src/gtk4-rs/gdk4-win32
# cargo test
  Downloaded fastrand v1.8.0
  Downloaded remove_dir_all v0.5.3
  Downloaded tempfile v3.3.0
  Downloaded gir-format-check v0.1.3
  Downloaded shell-words v1.1.0
  Downloaded 5 crates (61.9 KB) in 1.82s
   Compiling cfg-if v1.0.0
   Compiling fastrand v1.8.0
   Compiling shell-words v1.1.0
   Compiling gir-format-check v0.1.3
   Compiling remove_dir_all v0.5.3
   Compiling tempfile v3.3.0
   Compiling gdk4-win32 v0.5.0 (C:\Users\User\src\gtk4-rs\gdk4-win32)
error[E0412]: cannot find type `GdkWin32MessageFilterReturn` in this scope
   --> gdk4-win32\tests\abi.rs:206:25
    |
206 |         size: size_of::<GdkWin32MessageFilterReturn>(),
...
```

Reply:
does it work if you change https://github.com/gtk-rs/gir/blob/74b6e47217b730b04d119cdb2c7bd3d4e9de906e/src/codegen/sys/tests.rs#L510 to
 `        writeln!(w, "    (\"{ctype}\", Layout {{size: size_of::<crate::{ctype}>(), alignment: align_of::<crate::{ctype}>()}}),",`

Reply:
why would that change be necessary?

So if I get rid of `#[cfg(target_os = "linux")]` in sys/tests, it seems I can run the tests fine.

Reply:
@bilelmoussaoui the trouble is more about the Gdk prefix.. no idea why it is not stripped

Reply:
> @bilelmoussaoui the trouble is more about the Gdk prefix.. no idea why it is not stripped

I can't understand why they need to be stripped?

Reply:
cargo test --target x86_64-pc-windows-gnu:

```
error[E0412]: cannot find type `GdkWin32MessageFilterReturn` in this scope
   --> gdk4-win32/tests/abi.rs:206:25
    |
206 |         size: size_of::<GdkWin32MessageFilterReturn>(),
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
   ::: /home/elmarco/src/gtk4-rs/gdk4-win32/src/auto/enums.rs:11:1
    |
11  | pub enum Win32MessageFilterReturn {
    | --------------------------------- similarly named enum `Win32MessageFilterReturn` defined here
    |
help: an enum with a similar name exists
    |
206 |         size: size_of::<Win32MessageFilterReturn>(),
    |                         ~~~~~~~~~~~~~~~~~~~~~~~~
help: consider importing one of these items
    |
5   | use crate::ffi::GdkWin32MessageFilterReturn;
    |
5   | use ffi::GdkWin32MessageFilterReturn;

```

Reply:
I suppose we can land this PR and figure out the -sys tests later, we don't have working tests for gdk-x11 neither for example; see https://github.com/gtk-rs/gtk4-rs/issues/39

cc @sdroege @GuillaumeGomez, would you have an idea why the error above is happening?

Reply:
Why are the sys tests in `tests/abi.rs` instead of `sys/tests/abi.rs`?

Reply:
good catch, no idea! now cargo test works :)